plugins/rapid7_vulndb/unit_test/test_search_db.py

```diff
@@ -0,0 +1,117 @@
+                },
+            ],
+        }
+        self.assertEqual(actual, expected)
+
+    @patch("requests.get", side_effect=mock_request)
+    def test_search_db_no_results(self, mock_req):
+        actual = self.action.run(
+            {Input.SEARCH: self.params.get("search_no_results"), Input.DATABASE: self.params.get("database")}
+        )
+        expected = {"results_found": False, "search_results": []}
+        self.assertEqual(actual, expected)
+
+    @patch("requests.get", side_effect=mock_request)
+    def test_search_db_vulnerability_nexpose(self, mock_req):
+        actual = self.action.run(
+            {Input.SEARCH: self.params.get("search"), Input.DATABASE: self.params.get("vulnerability_database")}
+        )
+        expected = {
+            "results_found": True,
+            "search_results": [
+                {
+                    "identifier": "test_identifier_6",
+                    "title": "test_title_6",
+                    "published_at": "2019-01-01T00:00:00.000Z",
+                    "link": "https://vdb-kasf1i23nr1kl2j4.rapid7.com/v1/content/test_identifier_6",
+                    "solutions": "test_solution_1,test_solution_2",
+                }
+            ],
+        }
+        self.assertEqual(actual, expected)
+
+    @patch("requests.get", side_effect=mock_request)
+    def test_search_db_metasploit(self, mock_req):
+        actual = self.action.run(
+            {Input.SEARCH: self.params.get("search"), Input.DATABASE: self.params.get("metasploit_database")}
+        )
+        expected = {
+            "results_found": True,
+            "search_results": [
+                {
+                    "identifier": "test_identifier_5",
+                    "title": "test_title_5",
+                    "published_at": "2018-01-01T00:00:00.000Z",
+                    "link": "https://vdb-kasf1i23nr1kl2j4.rapid7.com/v1/content/test_identifier_5",
+                    "solutions": None,
+                }
+            ],
+        }
+        self.assertEqual(actual, expected)
+
+    @patch("requests.get", side_effect=mock_request)
+    def test_search_db_404(self, mock_req):
+        expected = "The requested resource could not be found."
+        with self.assertRaises(PluginException) as exception:
+            actual = self.action.run(
+                {Input.SEARCH: self.params.get("search_404"), Input.DATABASE: self.params.get("database")}
+            )
+        self.assertEqual(exception.exception.cause, expected)
+
+    @patch("requests.get", side_effect=mock_request)
+    def test_search_db_504(self, mock_req):
+        expected = "Server error occurred"
+        with self.assertRaises(PluginException) as exception:
+            actual = self.action.run(
+                {Input.SEARCH: self.params.get("search_504"), Input.DATABASE: self.params.get("database")}
+            )
+        self.assertEqual(exception.exception.cause, expected)
```

Comment:
Could you please parameterize and connect that tests together?

Reply:
Updated, let me know if there is another preferred method of doing so.